analyses/cell-type-ewings/exploratory_analysis/08-merged-celltypes.Rmd

```diff
@@ -0,0 +1,694 @@
+  "aucell results file does not exist" = file.exists(aucell_results_file)
+)
+```
+
+
+```{r, message=FALSE}
+# read in sce
+sce <- readr::read_rds(sce_file)
+
+# read in workflow results
+singler_df <- singler_results_files |> 
+  purrr::set_names(library_ids) |>
+  purrr::map(readr::read_tsv) |> 
+  dplyr::bind_rows(.id = "library_id")
+
+aucell_df <- readr::read_tsv(aucell_results_file) |> 
+  tidyr::separate(barcodes, "-", into = c("library_id", "barcodes"))
+
+# consensus cell types 
+consensus_df <- consensus_results_files |> 
+  purrr::map(readr::read_tsv) |> 
+  dplyr::bind_rows()
+
+# read in marker genes and combine into one list 
+visser_markers_df <- readr::read_tsv(visser_marker_genes_file) |> 
+  dplyr::select(cell_type, ensembl_gene_id, gene_symbol) |> 
+  unique()
+  
+cell_state_markers_df <- readr::read_tsv(cell_state_genes_file) |> 
+  dplyr::select(cell_type = cell_state, ensembl_gene_id, gene_symbol)
+
+all_markers_df <- dplyr::bind_rows(list(visser_markers_df, cell_state_markers_df))
+```
+
+## Prepare data for plotting
+
+```{r}
+all_results_df <- prep_results(
+  sce, 
+  singler_df = singler_df, 
+  cluster_df = NULL, 
+  aucell_df = aucell_df,
+  consensus_df = consensus_df,
+  cluster_nn = params$cluster_nn,
+  cluster_res = params$cluster_res,
+  join_columns = c("barcodes", "library_id")
+  ) |>
+  dplyr::mutate(barcodes = glue::glue("{library_id}-{barcodes}"))
+  
+cell_types <- unique(all_markers_df$cell_type)
+
+# get the mean expression of all genes for each cell state
+gene_exp_df <- cell_types |>
+  purrr::map(\(type){
+    calculate_mean_markers(all_markers_df, sce, type, cell_type)
+  }) |>
+  purrr::reduce(dplyr::inner_join, by = "barcodes")
+
+all_info_df <- all_results_df |> 
+  dplyr::left_join(gene_exp_df, by = "barcodes") |> 
+  dplyr::mutate(
+    singler_annotation = dplyr::if_else(is.na(singler_annotation), "unknown", singler_annotation)
+  )
+```
+
+## Combine classifications from `SingleR` and consensus cell types 
+
+The first thing we will do is compare cell type annotations obtained by each method, `SingleR` with tumor cells as a reference and consensus annotations. 
+
+Let's see how similar these annotations are and see if we can create a combined annotation. 
+Note that `SingleR` will label tumor cells but the consensus cell types only label normal cells. 
+Consensus cell types are observed when cell types from `SingleR` and `CellAssign` (using only normal tissue references) share a common ancestor. 
+If no consensus is found, the cells are labeled with "Unknown". 
+
+Any cells that are unable to be labeled via consensus cell types are labeled as "Unknown" and I expect these will line up with cells labeled as tumor cells by `SingleR`.  
+
+```{r, fig.height=10}
+# first get jaccard index between consensus and singler
+jaccard_mtx <- make_jaccard_matrix(
+  all_info_df,
+  "consensus_annotation",
+  "singler_annotation"
+)
+
+ComplexHeatmap::Heatmap(
```

Comment:
This heatmap is a really helpful addition, thanks! 
One thing I'm wondering that might be a bad idea is to only show _rows_ (aka SingleR annotations) which are non-0 in the heatmap to make it easier to see where values lineup. But, are any _actually_ 0 or are they just too faint to see?

Reply:
Hm it strikes me that this might be the same concept as when I was making deconvolution barplots that didn't show "other," and I was the only one who liked those plots 😂 
So, very much at your discretion here!